File: OSCommon/org/optimizationservices/oscommon/datastructure/osuri/URI.cs
using System;

namespace org.optimizationservices.oscommon.datastructure.osuri{
	/// <summary>
	/// A data structure class that corresponds to an xml element in the OSuL schema. 
	/// @author Jun Ma	
	/// @version 1.0, 09/01/2005
	/// @since OS 1.0
	/// @copyright (c) 2005
	/// </summary>
	public class URI{

		public string serviceName = "";
	
		public string matchType = "exact";
	
		public bool licenseRequired = false;
	
		public bool userNameRequired = false;
	
		public bool passwordRequired = false;

		public string value = "";

	}//class URI
}//namespace

File: OSCommon/src/org/optimizationservices/oscommon/datastructure/osuri/URI.java
/**
 * @(#)URI 1.0 03/14/2004
 *
 * Copyright (c) 2004
 */
package org.optimizationservices.oscommon.datastructure.osuri;



/**
 *
 * <p>A data structure class that corresponds to an xml element in the OSqL schema.  
 *  
 * </p>
 * @author Robert Fourer, Jun Ma, Kipp Martin
 * @version 1.0, 03/14/2004
 * @since OS 1.0
*/
public class URI{
	
	public String serviceName = "";
	
	public String matchType = "exact";
	
	public boolean licenseRequired = false;
	
	public boolean userNameRequired = false;
	
	public boolean passwordRequired = false;

	public String value = "";
	
}//class URI
